### DISCUSSION TITLE
gocryptfs -init -xchacha: Cannot read decrypted files on cppcryptfs

### DISCUSSION THREAD

**Discussion by magnemann** [rfjakob/gocryptfs#720] [Q&A] [answered]
Hi,
On Archlinux, I init a new fs on a ntfs-partition.

> gocryptfs -init -xchacha -deterministic-names -raw64 /run/media/magne/tbox/diamant-magne/

The config look like this:

> {
> 	"Creator": "gocryptfs v2.3",
> 	"EncryptedKey": "*thekeygoeshere*",
> 	"ScryptObject": {
> 		"Salt": "*thesaltgoeshere*",
> 		"N": 65536,
> 		"R": 8,
> 		"P": 1,
> 		"KeyLen": 32
> 	},
> 	"Version": 2,
> 	"FeatureFlags": [
> 		"HKDF",
> 		"XChaCha20Poly1305",
> 		"EMENames",
> 		"LongNames",
> 		"Raw64"
> 	]
> }

Then I mount it:

> magne@rawjo /r/m/m/t/diamant-magne> gocryptfs /run/media/magne/tbox/diamant-magne/ /mnt/gm-clear/
> Password: 
> Decrypting master key
> Filesystem mounted and ready.

Then I copy some files to it, and all seems normal. I can read and modify the files.

The the scary part starts. I switch to windows10 and fire up cppcryptfs.
When I try to mount, I get
![image](https://user-images.githubusercontent.com/93428253/224507063-76dc8dfc-1b8c-466e-b41a-d845faffc368.png)
So I remove that flag in config file and try again.
I get:
![image](https://user-images.githubusercontent.com/93428253/224507148-6da073af-911c-414b-9775-f8cbf4f9c315.png)
I add the flag and finally it mounts. I can see the files and the decrypted filenames.
But when I try to open a file i get read error, depending on filetype:
![image](https://user-images.githubusercontent.com/93428253/224507250-40f6405c-9509-455a-9144-e8d99680377c.png)
![image](https://user-images.githubusercontent.com/93428253/224507275-f285fb1b-6b04-4af6-a811-1f1fc7960d21.png)

Does anyone have a tip for me? Thanks a lot. 😃 

Read Error.
Data could not be read from the file.

**Comment by rfjakob** [comment, marked as answer]
Looks like cppcryptfs does not support `-xchacha`. You need to run `gocryptfs -init` without that flag.

**Reply by rfjakob** [reply]
Also, you should rather not mess around with your `gocryptfs.conf` like that :)

Doing that means that `gocryptfs.conf` does not match the actually encrypted files anymore, which is NOT SO GOOD (tm).

**Reply by magnemann** [reply]
Tried again without `-xchacha`, works like a charm now!
I thought `gocryptfs.conf` contained options on how to mount. I try to keep my fingers off that file.
Thanks!

**Reply by rfjakob** [reply]
So, gocryptfs.conf has the info on how the files are encrypted. Any mount options (like `--quiet` etc) must be passed on the command line each time, these are not saved.
